action-server/covidflow/actions/daily_ci_feel_better_form.py

```diff
@@ -1,222 +1,186 @@
-
-    ## override to play initial message and set feel_worse slot
-    async def _activate_if_required(
-        self,
-        dispatcher: "CollectingDispatcher",
-        tracker: "Tracker",
-        domain: Dict[Text, Any],
+
+        template_name = f"utter_ask_{FORM_NAME}_{HAS_COUGH_SLOT}"
+
+        if tracker.get_slot(LAST_HAS_COUGH_SLOT) is True:
+            dispatcher.utter_message(template=f"{template_name}___still")
+        else:
+            dispatcher.utter_message(template=template_name)
+
+        return []
+
+
+class ActionAskDailyCiFeelBetterFormHasOtherMildSymptoms(Action):
+    def name(self) -> Text:
+
+        return ASK_HAS_OTHER_MILD_SYMPTOMS_ACTION_NAME
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
     ) -> List[EventType]:
-        events = []
+        bind_logger(tracker)
 
-        if tracker.active_loop.get("name") != FORM_NAME:
-            dispatcher.utter_message(
-                template="utter_daily_ci__feel_better__acknowledge"
-            )
+        template_name = f"utter_ask_{HAS_OTHER_MILD_SYMPTOMS_SLOT}"
 
-            events.append(SlotSet(FEEL_WORSE_SLOT, False))
+        if tracker.get_slot(LAST_SYMPTOMS_SLOT) == Symptoms.MODERATE:
+            dispatcher.utter_message(template=f"{template_name}___with_acknowledge")
+        else:
+            dispatcher.utter_message(template=template_name)
+
+        return []
 
-        return await super()._activate_if_required(dispatcher, tracker, domain) + events
 
-    @staticmethod
-    def required_slots(tracker: Tracker) -> List[Text]:
-        """A list of required slots that the form has to fill"""
+class ValidateDailyCiFeelBetterForm(Action):
+    def name(self) -> Text:
 
-        slots = [HAS_FEVER_SLOT, HAS_COUGH_SLOT]
+        return VALIDATE_ACTION_NAME
 
-        last_symptoms = tracker.get_slot(LAST_SYMPTOMS_SLOT)
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[EventType]:
+        bind_logger(tracker)
 
-        if last_symptoms == Symptoms.MODERATE:
-            slots.append(HAS_DIFF_BREATHING_SLOT)
+        extracted_slots: Dict[Text, Any] = tracker.form_slots_to_validate()
+
+        validation_events: List[EventType] = []
+        for slot_name, slot_value in extracted_slots.items():
+            slot_events = [SlotSet(slot_name, slot_value)]
+
+            if slot_name == HAS_FEVER_SLOT:
+                slot_events += _validate_has_fever(slot_value, dispatcher)
+            elif slot_name == HAS_COUGH_SLOT:
+                slot_events += _validate_has_cough(slot_value, dispatcher)
+                if (
+                    tracker.get_slot(LAST_SYMPTOMS_SLOT) != Symptoms.MODERATE
+                ):  # Next slot only happens when symptoms are moderate
+                    slot_events.append(
+                        SlotSet(HAS_DIFF_BREATHING_SLOT, SKIP_SLOT_PLACEHOLDER)
+                    )
```

Comment:
Ton commentaire est mélangeant, et semble contredire la condition juste au dessus.

Reply:
"is only asked"? j'avoue que "happens" c'est pas particulièrement précis XD

Reply:
Mais c'est quoi le next slot ?

Reply:
"Is only asked" serait un peu mieux, oui.